src/python/pants/backend/native/tasks/native_external_library_fetch.py

```diff
@@ -129,40 +136,35 @@ def execute(self):
       with self.invalidated(native_lib_tgts,
                             invalidate_dependents=True) as invalidation_check:
         resolve_vts = VersionedTargetSet.from_versioned_targets(invalidation_check.all_vts)
-        vts_results_dir = self._prepare_vts_results_dir(resolve_vts)
         if invalidation_check.invalid_vts or not resolve_vts.valid:
           for vt in invalidation_check.all_vts:
-            self._fetch_packages(vt, vts_results_dir)
+            self._fetch_packages(vt)
 
-        native_external_libs_product = self._collect_external_libs(vts_results_dir)
+        native_external_libs_product = self._collect_external_libs(invalidation_check.all_vts)
         self.context.products.register_data(NativeExternalLibraryFiles,
                                             native_external_libs_product)
```

Comment:
I think the above sounds like a good idea -- unsure if we would need to override the `copy()` method to return a `NativeExternalLibrary` instead of a `UnionProducts`, because I think the `UnionProducts` method would take precedence over the datatype version.